client/core/core.go

```diff
@@ -932,10 +937,35 @@ func (dc *dexConnection) reconcileTrades(srvOrderStatuses []*msgjson.OrderStatus
 			var oid order.OrderID
 			copy(oid[:], srvOrderStatus.ID)
 			trade := knownActiveTrades[oid] // no need to lock dc.tradeMtx
+			if trade == nil {
+				dc.log.Warnf("Server reported status for order %v that we did not request.", oid)
+				continue
+			}
 			trade.mtx.Lock()
 			updateOrder(trade, srvOrderStatus)
 			trade.mtx.Unlock()
 		}
+
+		// Treat orders with no status reported as revoked.
+	reqsLoop:
+		for _, req := range orderStatusRequests {
+			for _, res := range orderStatusResults {
+				if req.OrderID.Equal(res.ID) {
+					continue reqsLoop
+				}
+			}
+			// No result for this order.
+			dc.log.Warnf("Server did not report status for order %v", req.OrderID)
```

Comment:
I believe we've encountered cases where non-existent (server-side) orders would be handled by this.  I'm not sure how a client would have an order but not server, but I think this has happened and it should be handled.